test/deserialize/record.js

```diff
@@ -183,3 +183,38 @@ it('should use the result of the migrate method', (test) => {
     },
   })
 })
+
+
+it('should apply deserializer onto outcoming data if option is present', (test) => {
+  const SampleRecord = immutable.Record({
+    'a': 1,
+    'b': 2,
+  }, 'SampleRecord')
+
+  const data = {
+    '__record': 'SampleRecord',
+    'data': {
+      'a': 5,
+      'b': 6,
+    },
+  }
+
+  const expectedResult = SampleRecord({
+    'a': '5-transformed',
+    'b': '6-transformed'
+  })
+
+  helpers.testDeserialization(test, data, expectedResult, {
+    recordTypes: {
+      'SampleRecord': SampleRecord,
+    },
+    deserializers: { 
+      'SampleRecord': (data) => {
+        return Object.assign(data, {
+          'a': `${data['a']}-transformed`,
+          'b': `${data['b']}-transformed`,
+        }) 
+      },
+    },
+  })
+})
```

Comment:
It would make sense to add a test which checks that the lack of deserializers for **all** record types does not break the deserialization.

Sth like deserializing two records—of types `A` and `B`—while only providing a deserializer for `A`.

Reply:
I would also try deserializing non-record structures with the `deserializers` option being set.